cloudencrypt/dual.go

```diff
@@ -0,0 +1,96 @@
+package cloudencrypt
+
+import (
+	"context"
+)
+
+const (
+	mapKeyEncryptedKey   = "key"
+	mapKeyEncryptedValue = "value"
+)
+
+// DualEncryptor implements Encryptor by encrypting the value using the NativeEncryptor
+// and then encrypting the secret key using the provided encryptor.
+type DualEncryptor struct {
+	encryptor Encryptor
+}
+
+func NewDualEncryptor(ctx context.Context, encryptor Encryptor) (*DualEncryptor, error) {
+	return &DualEncryptor{
+		encryptor: encryptor,
+	}, nil
+}
+
+func (encryptor *DualEncryptor) Encrypt(ctx context.Context, value []byte, metadata ...MetadataKV) ([]byte, error) {
```

Comment:
```suggestion
func (dualEncryptor *DualEncryptor) Encrypt(ctx context.Context, value []byte, metadata ...MetadataKV) ([]byte, error) {
```

Reply:
I'm against this.

Reply:
What about `(de *DualEncryptor)` ?  https://github.com/prometheus/client_golang/blob/main/prometheus/labels.go#L66

Reply:
I really don't wanna use different name in each implementation.